applications/DeepSpeed-Chat/training/utils/data/data_utils.py

```diff
@@ -286,6 +287,33 @@ def create_prompt_dataset(local_rank,
             eval_dataset = ConcatDataset(eval_datasets)
             shuffle_idx = get_shuffle_idx(seed, eval_size)
             eval_dataset = Subset(eval_dataset, shuffle_idx.tolist())
+
+        # Append the SFT-only dataset if it exists, and current phase is 1(SFT).
+        if train_phase == 1 and sft_only_data_path:
```

Comment:
for this sft_only_data_path, please add it as part of the string fname = f"{fname}_split{data_split}_phase{train_phase}_seed{seed}_tokenizer{tokenizer_name}_seqlen{max_seq_len}" above. This is because we are actually caching the processed dataset to avoid processing time every time. If you don't add sft_only_data_path as part of the cache file name, then when you change SFT-only dataset you will get a wrong cached data.